client/src/components/NewSurveyModal.js

```diff
@@ -1,22 +1,53 @@
-import React from "react";
+import React, { useState, useCallback } from "react";
 
-import { Modal, TextContainer } from "@shopify/polaris";
+import { Modal, FormLayout, TextField } from "@shopify/polaris";
 
 export function NewSurveyModal({ open, openHandler }) {
+  const [name, setName] = useState("");
+
+  const handleTitleChange = useCallback(value => setName(value), []);
+
+  const handleModalAction = () => {
+    if (name === "") {
+      openHandler();
+      return;
+    }
+
+    async function submit() {
+      const res = await fetch(`/new?name=${name}`, {
+        method: "POST"
```

Comment:
I guess spring would auto handle this with @RequestParam but shouldn't params be provided in the body for post requests?

Reply:
We don't have our [controller](https://github.com/vikrambombhi/SimpleSurvey/blob/master/src/main/java/com/example/survey/models/SurveyController.java#L33) set up correctly to accept It as a body - [context](https://stackoverflow.com/questions/57623466/spring-boot-rest-required-string-parameter-is-not-present).

Reply:
see #33